File: 프로그래머스/3/42579. 베스트앨범/베스트앨범.py
def solution(genres, plays):
    
    answer = []
    songs = {}
    total = {}
    
    for i in range(len(genres)):
        if genres[i] in songs:
            songs[genres[i]].append((i,plays[i]))
            total[genres[i]] += plays[i]
        else:
            songs[genres[i]] = [(i, plays[i])]
            total[genres[i]] = plays[i]
    
    total_sort = sorted(total.items(), key = lambda x : -x[1])

    for genre, count in total_sort:
        songs_sort = sorted(songs[genre], key = lambda x: -x[1])
        for x, y in songs_sort[:2]:
            answer.append(x)
    
    return answer

File: 프로그래머스/3/42579. 베스트앨범/베스트앨범.js
function solution(genres, plays) {
    
    let songs = {}
    let total = {}
    var answer = [];
    
    for (let i = 0; i < genres.length; i++) {
        if (genres[i] in songs) {
            songs[genres[i]].push([i, plays[i]])
            total[genres[i]] += plays[i]
        } else {
            songs[genres[i]] = [[i, plays[i]]]
            total[genres[i]] = plays[i]
        }
    }
    
    const total_sort = Object.entries(total).sort((a, b) => b[1] - a[1])
    total_sort.forEach(x => {
       songs[x[0]].sort((a, b) => b[1] - a[1]).slice(0, 2).map(x => {answer.push(x[0])} )
    })
    
    return answer;
}